File: haushaltsbuch/src/main/java/org/github/haschi/haushaltsbuch/infrastruktur/modellierung/de/_Aggregatkennung.java
package org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import org.immutables.value.Value;

import java.util.UUID;

@Value.Immutable
@Eingehüllt
@JsonSerialize(using = AggregatkennungSerialisierer.class, as = Aggregatkennung.class)
@JsonDeserialize(using = AggregatkennungDeserialisierer.class, as = Aggregatkennung.class)
public abstract class _Aggregatkennung extends Umhüller<UUID>
{
    public static Aggregatkennung neu()
    {
        return Aggregatkennung.of(UUID.randomUUID());
    }

    public static Aggregatkennung of(final String id)
    {
        return Aggregatkennung.of(UUID.fromString(id));
    }
}

File: haushaltsbuch/src/main/kotlin/org/github/haschi/haushaltsbuch/infrastruktur/modellierung/de/Aggregatkennung.kt
package org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize

import java.util.UUID

@JsonSerialize(using = AggregatkennungSerialisierer::class)
@JsonDeserialize(using = AggregatkennungDeserialisierer::class)
data class Aggregatkennung(val id: UUID) {

    companion object {
        fun neu(): Aggregatkennung {
            return Aggregatkennung(UUID.randomUUID())
        }

        fun aus(id: String): Aggregatkennung {
            return Aggregatkennung(UUID.fromString(id))
        }
    }
}
